source/clear-linux/tutorials/spark.rst

```diff
@@ -0,0 +1,136 @@
+  .. _spark:
+
+Set up a standalone cluster system using Apache\* Spark\*
+#########################################################
+
+This tutorial describes how to install, configure, and run Apache Spark on
+Clear Linux OS for Intel(r) Architecture. Apache Spark is a fast
+general-purpose cluster computing system with the following features:
+
+*  Provides high-level APIs in Java\*, Scala\*, Python\*, and R\*.
+*  Includes an optimized engine that supports general execution graphs.
+*  Supports high-level tools including Spark SQL, MLlib, GraphX, and Spark
+   Streaming.
+
+In this tutorial, you will install Spark on a single machine running the
+master daemon and a worker daemon.
+
+Prerequisites
+*************
+
+This tutorial assumes you have installed |CL| on your host system.
+For detailed instructions on installing |CL| on a bare metal system, visit
+the :ref:`bare metal installation tutorial<bare-metal-install>`.
+
+Before you install any new packages, update |CL| with the following command:
+
+.. code-block:: bash
+
+   sudo swupd update
+
+Install Apache Spark
+********************
+
+Apache Spark is included in the :file:`big-data-basic` bundle. To install the
+framework, enter:
+
+.. code-block:: bash
+
+   sudo swupd bundle-add big-data-basic
+
+Configure Apache Spark
+**********************
+
+#. Create the configuration directory with the command:
+
+   .. code-block:: bash
+
+      sudo mkdir /etc/spark
+
+#. Copy the default templates from :file:`/usr/share/defaults/spark` to
+   :file:`/etc/spark` with the command:
+
+   .. code-block:: bash
+
+      sudo cp /usr/share/defaults/spark/* /etc/spark
+
+   .. note:: Since |CL| is a stateless system, you should never modify the
+      files under the :file:`/usr/share/defaults` directory. The software
+      updater overwrites those files.
+
+
+#. Copy the template files below to create custom configuration files:
+
+   .. code-block:: bash
+
+      sudo cp /etc/spark/spark-defaults.conf.template /etc/spark/spark-defaults.conf
+      sudo cp /etc/spark/spark-env.sh.template /etc/spark/spark-env.sh
+      sudo cp /etc/spark/log4j.properties.template /etc/spark/log4j.properties
+
+#. Edit the :file:`/etc/spark/spark-env.sh` file and add the
+   :option:`SPARK_MASTER_HOST` variable. Replace the example address below
+   with your localhost IP address. (View your IP address using the
```

Comment:
Do not use parenthesis. Remove them or make the line a note. The use of parenthesis is ambiguous and should be avoided.